core/state_processor.go

```diff
@@ -209,3 +213,28 @@ func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *stat
 	_, _, _ = vmenv.Call(vm.AccountRef(msg.From), *msg.To, msg.Data, 30_000_000, common.U2560)
 	statedb.Finalise(true)
 }
+
+// ProcessParentBlockHash stores the parent block hash in the history storage contract
+// as per EIP-2935.
+func ProcessParentBlockHash(prevHash common.Hash, vmenv *vm.EVM, statedb *state.StateDB) {
+	if vmenv.Config.Tracer != nil && vmenv.Config.Tracer.OnSystemCallStart != nil {
+		vmenv.Config.Tracer.OnSystemCallStart()
+	}
+	if vmenv.Config.Tracer != nil && vmenv.Config.Tracer.OnSystemCallEnd != nil {
+		defer vmenv.Config.Tracer.OnSystemCallEnd()
+	}
```

Comment:
```suggestion
	if vmenv.Config.Tracer != nil {
		if Config.Tracer.OnSystemCallStart != nil {
			vmenv.Config.Tracer.OnSystemCallStart()
		}
		if vmenv.Config.Tracer.OnSystemCallEnd != nil {
			defer vmenv.Config.Tracer.OnSystemCallEnd()
		}
	}
```
IMO it's nicer to clump trace-code into one standalone scope. Yeah I know it's a copy-paste of the code above, so I'd prefer it to be like this in both places. I'm not adamant, just a suggestion though

Reply:
Agreed it is more readable. Fixed in both 👍